### PR TITLE
Add guidance how to write tests

### PR DESCRIPTION
**Pull request by hhorak**
I've tried to write-up what I'd like to say every image maintainers when he asks about how to create tests for container images. I'm not sure whether this is the form that is expected for this guide, but we can work on it.

### CODE REVIEW

File: testing/testing_index.adoc [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+:toclevels 4:
+:homepage https://github.com/projectatomic/container-best-practices:
+
+=== Purpose of the testing
+
+Container images generally consist of distro packages and some scripts that help to start the container properly. For example in case of MariaDB database, it is set of RPMs with mariadb-server that provides the main functionality, but we need some scripts that handle proper database stack initialization, changes basic configuration based on the container run options etc.
+
+If we want to test basic functionality of the container, we do not have to test the RPM functionality, because it should have already be done during RPM develpment process. We need to focus on testing the additional scripts instead and test the API of the container -- whether it works as it is described. Again, in case of the MariaDB database, we do not run MariaDB unit test in the container, but we focus on whether the database is initialized properly, whether configuration is changed after proper options are set on container run command etc.
+
+=== Converntions for test scripts
```
Comment (bexelbie) [inline testing/testing_index.adoc]: Conventions for Test Scripts

File: testing/testing_index.adoc [hunk 2]
```diff
@@ -0,0 +1,97 @@
+// vim: set syntax=asciidoc:
+[[Testing]]
+== Testing
+:data-uri:
+:toc:
+:toclevels 4:
+:homepage https://github.com/projectatomic/container-best-practices:
+
+=== Purpose of the testing
```
Comment (bexelbie) [inline testing/testing_index.adoc]: What Should be Tested

File: testing/testing_index.adoc [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+Container images generally consist of distro packages and some scripts that help to start the container properly. For example in case of MariaDB database, it is set of RPMs with mariadb-server that provides the main functionality, but we need some scripts that handle proper database stack initialization, changes basic configuration based on the container run options etc.
```
Comment (bexelbie) [inline testing/testing_index.adoc]: Container images generally consist of distribution packages and some scripts that help start the container properly. For example, a container with a MariaDB database typically consists of a set of RPMs, such as mariadb-server, that provides the main functionality and some scripts that handle initialization, setup, etc.

File: testing/testing_index.adoc [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+If we want to test basic functionality of the container, we do not have to test the RPM functionality, because it should have already be done during RPM develpment process. We need to focus on testing the additional scripts instead and test the API of the container -- whether it works as it is described. Again, in case of the MariaDB database, we do not run MariaDB unit test in the container, but we focus on whether the database is initialized properly, whether configuration is changed after proper options are set on container run command etc.
```
Comment (bexelbie) [inline testing/testing_index.adoc]: If we want to test the basic functionality of the container, we do not have to test the RPM functionality.  The benefit of using the distribution packaging is that we know testing has already been done during the RPM develpment process.  Instead, we need to focus on testing the added scripts instead and the API of the container.  The goal is to determine if it works as described. For example, a MariaDB database container, we do not run the MariaDB unit tests in the container.  Instead we focus on whether the database is initialized, configured, and responds to commands properly.

File: testing/testing_index.adoc (hunk 1 above)
Comment (baude) [inline testing/testing_index.adoc]: Any advice on how to automate this?

File: testing/testing_index.adoc (hunk 3 above)
Comment (baude) [inline testing/testing_index.adoc]: @hhorak do you think showing the mariadb dockerfile would be helpful to create context here? or perhaps portions of it?

File: testing/testing_index.adoc [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+=== Converntions for test scripts
+
+It is a good practice to keep the basic sanity test for the image together with the image sources. For example test/run might be a script, that tests image specified by IMAGE_NAME environment variable.
```
Comment (bexelbie) [inline testing/testing_index.adoc]: It is good practice to keep the basic sanity tests for the image together with the image sources. For example test/run might be a script, that tests the image specified by the IMAGE_NAME environment variable.

Comment (bexelbie) [inline testing/testing_index.adoc]: This paragraph could use some expansion, specifically on why IMAGE_NAME should be used

File: testing/testing_index.adoc [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+=== Examples of test scripts
+
+Some examples of test scripts might be found in the container images for OpenShift, for example:
```
Comment (bexelbie) [inline testing/testing_index.adoc]: These examples of test scripts can be found in the container images for OpenShift:

File: testing/testing_index.adoc [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+* https://github.com/openshift/postgresql/blob/master/9.4/test/run
+* https://github.com/openshift/sti-python/blob/master/3.4/test/run
+
+The minimum script that verifies a container image by running it as deamon and then running some script that checks the proper functionality, is shown bellow. It stores IDs of containers created during the test into a temporary directory, which helps to remove the contaiers after test finishes.
```
Comment (bexelbie) [inline testing/testing_index.adoc]: A minimal script that verifies a container image by running it as daemon and then running a script that checks the proper functionality, is shown below. It stores the IDs of the containers created during the test in a temporary directory.  This makes it easy to clean up those containers after the test finishes.

**Comment by bexelbie** [comment]
I added mostly language edits.

**Comment by hhorak** [comment]
Sorry for shameful delay, finally got to fix the PR and include suggested edits.

**Comment by eliskasl** [comment]
@hhorak, can you please link the new chapter from the index.adoc file so it shows up in the book? thanks

**Comment by hhorak** [comment]
@eliskasl Done.

**Pull request merged**